File: javascript/challenge-2025-08-11.js

function countVowels(s) {
  const vowels = ["a", "e", "i", "o", "u"];
  s = s.toLowerCase();
  let result = 0;
  for (const x of s) {
    if (vowels.includes(x)) {
      result += 1;
    }
  }
  return result;
}

function isBalanced(s) {
  const n = s.length;
  const middle = Math.floor(n/2);
  const half_1 = s.slice(0, middle);
  const half_2 = s.slice(-middle);
  const n_vowels_1 = countVowels(half_1);
  const n_vowels_2 = countVowels(half_2);
  const result = n_vowels_1 == n_vowels_2;
  console.log("result:", result);
  return result;
}

File: python/challenge-2025-08-11.py

def countVowels(s):
    result = 0
    vowels = ["a", "e", "i", "o", "u"]
    s = s.lower()
    for x in s:
        if x in vowels:
            result += 1
    return result

def is_balanced(s):
    n = len(s)
    middle = n//2 
    half_1 = s[:middle]
    half_2 = s[-middle:]
    n_vowels_1 = countVowels(half_1)
    n_vowels_2 = countVowels(half_2)
    result = n_vowels_1 == n_vowels_2
    print(f"result: {result}")
    return result



